example.cpp

```diff
@@ -246,21 +246,27 @@ TEST_CASE_METHOD(CoherentUnitTestFixture, "test_grover")
                  "should output 00100110 11011001.)"
               << std::endl;
 
+    unsigned char toSearch[256];
+    for (i = 0; i < 256; i++) {
+        toSearch[i] = 255 - i;
+    }
+
     qftReg->SetPermutation(0);
     qftReg->SetBit(16, true);
-    qftReg->H(0, 8);
+    qftReg->H(8, 8);
+    qftReg->SuperposeReg8(8, 0, toSearch);
 
     // Twelve iterations maximizes the probablity for 256 searched elements.
     for (i = 0; i < 12; i++) {
         qftReg->DEC(100, 0, 8);
         qftReg->SetZeroFlag(0, 8, 16);
         qftReg->INC(100, 0, 8);
-        qftReg->H(0, 8);
-        qftReg->SetZeroFlag(0, 8, 16);
-        qftReg->H(0, 8);
+        qftReg->EntangledH(8, 0, 8);
+        qftReg->SetZeroFlag(8, 8, 16);
+        qftReg->EntangledH(8, 0, 8);
         qftReg->Z(16);
-        std::cout << "Iteration " << i
-                  << ", chance of 100:" << (qftReg->ProbAll(100) + qftReg->ProbAll(100 + (1 << 16))) << std::endl;
+        std::cout << "Iteration " << i << ", chance of match:"
+                  << (qftReg->ProbAll(100 + (155 << 8)) + qftReg->ProbAll(100 + (155 << 8) + (1 << 16))) << std::endl;
```

Comment:
Probably worth a constant or set of constants at the top of the test to pull these magic numbers out of the middle (and bottom).